bokeh/models/annotations.py

```diff
@@ -439,9 +439,10 @@ class Tooltip(Annotation):
     """
     level = Override(default="overlay")
 
-    side = Either(Auto, Enum(Side), default="auto", help="""
+    attachment = Enum("horizontal", "vertical", "left", "right", "above", "below", help="""
```

Comment:
I don't think the functionality belong here. But it could be appropriate for an annotation to express its preference for  "feel free to dodge me" or "don't dodge me" to the thing that goes.  

> On May 26, 2016, at 06:05, Mateusz Paprocki <notifications@github.com> wrote:
> 
> In bokeh/models/annotations.py:
> 
> > @@ -439,9 +439,10 @@ class Tooltip(Annotation):
> >      """
> >      level = Override(default="overlay")
> >  
> > -    side = Either(Auto, Enum(Side), default="auto", help="""
> > +    attachment = Enum("horizontal", "vertical", "left", "right", "above", "below", help="""
> that we might want to automatically dodge multiple tooltips
> 
> This functionality doesn't belong to an annotation, rather to a tool or perhaps a manager on top of that. Still auto in this case may make sense, but to automatically choose between horizontal and vertical dimensions.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
>